src/objectivec/oga_internal.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "cxx_api.h"
+#import "error_utils.h"
+#import "ort_genai_objc.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface OGAModel ()
+
+- (const OgaModel&)CXXAPIOgaModel;
+
+@end
+
+@interface OGATokenizer ()
+
+- (const OgaTokenizer&)CXXAPIOgaTokenizer;
+
+@end
+
+@interface OGASequences ()
+
+- (nullable instancetype)initWithError:(NSError**)error;
+- (instancetype)initWithCXXPointer:(std::unique_ptr<OgaSequences>)ptr;
+
+- (OgaSequences&)CXXAPIOgaSequences;
+
+@end
+
+@interface OGAGeneratorParams ()
+
+- (OgaGeneratorParams&)CXXAPIOgaGeneratorParams;
+
+@end
+
+@interface OGAImages ()
+
+- (OgaImages*)CXXAPIOgaImages;
+
+@end
+
+@interface OGATensor ()
+
+- (instancetype)initWithCXXPointer:(std::unique_ptr<OgaTensor>)ptr;
+
+- (OgaTensor&)CXXAPIOgaTensor;
+
+@end
+
+@interface OGANamedTensors ()
+
+- (instancetype)initWithCXXPointer:(std::unique_ptr<OgaNamedTensors>)ptr
+    NS_DESIGNATED_INITIALIZER;
```

Comment:
should some of the other initializers in this file also be marked `NS_DESIGNATED_INITIALIZER`?

Reply:
yeah. I reorganized some of them initializers.